airflow/www/views.py

```diff
@@ -353,6 +354,8 @@ def set_overall_state(record):
                 'label': item.label,
                 'extra_links': item.extra_links,
                 'is_mapped': item.is_mapped,
+                'has_outlet_datasets': any(isinstance(i, Dataset) for i in getattr(item, "_outlets", [])),
```

Comment:
I wanted to mention this does not work with mapped operators (it does not have `_outlets`, only `outlets`), but then realised I don’t understand the difference between the two. Is the non-underscore-prefixed one populated by the other? Where?

Reply:
this is a known issue, [documented here](https://github.com/apache/airflow/projects/16#card-84459116), for someone to look into at some point

Reply:
So I just played with this a bit, and while you can set an outlet in partial, it doesn't actually function. (because it gets serialized into `outlets`?). We might need to investigate that sooner than later, but I'm not sure we should block this PR in the meantime so the UI effort can continue.

The getattr was because mapped operators don't have `_outlets`, so while this doesn't "work", this also doesn't "break" things further.

Reply:
Yeah I think when I implemented MappedOperator I just assumed `outlets` and `_outlets` (also inlets) are sync-ed and didn’t look deep into it. Now I did I don’t understand why there are two attributes to begin with and have no idea how things _should_ work.

Reply:
I suspect they are disconnected for no good reason (or no reason that exists anymore) and `_outlets` should just be removed (which we can do, as it's a private attribute after all)